### DISCUSSION TITLE
Pentair IC40 flow switch + ET4 - does anyone else have a problem with salt readings?

### DISCUSSION THREAD

**Discussion by johnny2678** [tagyoureit/nodejs-poolController#834] [Q&A]
seems like I am replacing my flow sensor every few months, as you can see from the flat readings below.  Does anyone else have this problem?

![image](https://github.com/tagyoureit/nodejs-poolController/assets/8219743/040388f7-0931-4c87-84d1-9eb02187a542)

Any chance this is related to ET4 and would be improved by converting to 100% nixie?  I'm still not sure I have the stomach for that but I'm working up to it.

**Comment by rstrouse** [comment]
Is the issue the flow or is it not reading the temperature properly when it fails?

**Comment by johnny2678** [comment]
flow is fine, status always reads 'OK'

not sure if it's reading temp since that's not reported anywhere but it's not throwing a cold water error.  I've always had problems with these.  In the winter I get cold water warnings all the time when the temp is around 75.  No idea why.

**Reply by rstrouse** [reply]
Check to make sure your cell doesn't have buildup on it first.  Then look at the flow switch for corrosion at the very top of it.  This is indicative of water intrusion into the copper parts.

EDIT:  I mean to look at the inside where the stainless flapper meets the plastic at the very top of it.

**Reply by johnny2678** [reply]
yea, every time I pull the cell it is pristine.  Probably because my acid injects just prior to the cell.  Flow switch is brand new (well, less than 2 months old) so there's no corrosion that I can see.  My bet would be on a comms issue before a hardware issue.  But I have no way to prove that.  

Just easier to blame ET4 because it keeps my brain primed to replace it with nixie some day. 🤣
